servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/SslAndNonSslConnectionsTest.java

```diff
@@ -185,6 +187,20 @@ public void singleAddressClientWithSslToSecureServer() throws Exception {
         }
     }
 
+    @Test
+    public void singleAddressClientWithSslToSecureServerWithoutPeerHost() throws Exception {
+        assert secureServerCtx != null;
+        try (BlockingHttpClient client = HttpClients.forSingleAddress(serverHostAndPort(secureServerCtx))
+                .sslConfig(new ClientSslConfigBuilder(DefaultTestCerts::loadServerCAPem)
+                        .peerHost(null)
+                        // if verification is not disabled, identity check fails against the undefined address
+                        .hostnameVerificationAlgorithm("")
```

Comment:
@Scottmitch if this approach also works we can consider removing `disableHostnameVerification()`. If we prefer `disableHostnameVerification()`, we should consider enforcing a non-empty verification algorithm. Having two ways to do the same thing is ambigious.

Reply:
I think the intention was to block users from disabling the verification, but I believe there are cases where it makes sense. So agree with @idelpivnitskiy, best to leave just one way to disable verification.

Reply:
my vote: remove `disableHostnameVerification` update javadocs for `hostnameVerificationAlgorithm`